tools/autograd/gen_python_functions.py

```diff
@@ -891,6 +891,14 @@ def get_py_formal_arg(arg):
             typename = 'TensorList[{}]'.format(len(typenames))
         else:
             typename = typenames[0]
+        # The nn module bindings are not exposed to the user directly
+        # but via torch.nn modules and functionals.
+        if len(output_args) == 1 and declaration['python_module'] != 'nn':
+            assert output_args[0]['name'] == 'out', \
+                "For function named {}: Our Python frontend uses out as" \
+                " the keyword argument name for output arguments.".format(name)
+        # NB: For more than 1 output args the type name is a TensorList
```

Comment:
is this note correct?  There are a bunch of cases with multiple output arguments that are Tensors AFAII.

Reply:
It's determined above. `typenames` is of the same length as `output_args` and `typename` is set to be `TensorList[len(typenames)]` if that's longer than 1.